File: Etudes/LOTS/projetLot/SCALA/src/informatique/Ordinateur.scala
package informatique
import general.Sommet

class Ordinateur extends Sommet{
  type U =Cable;
  type T =Reseau ;
  
  def addAret(aret: U): String = {
    arets.+:(aret);
    return "OK\n";
  }

  def setGraphe(graphe: T): String = {
    if (isNullGraphe()) {
      this.graphe = graphe;
      graphe.sommets.append(this);
      return "OK\n";
    }
    return "NoN OK\n";
  }
}

File: Etudes/LOTS/projetLot/CSharp/CSharp/Informatique/Ordinateur.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;

using General;

namespace Informatique
{

  public class Ordinateur : Sommet
  {
      public Ordinateur(String name) {
		this.name=name;
	}

	override public bool compatibleGraphe(Graphe graphe) {
		return (graphe is Reseau);
	}

	override public bool compatibleAret(Aret aret) {
		return (aret is Cable);
	}
  }

}  // end of namespace Informatique

